src/main/java/space/obminyashka/items_exchange/dao/AdvertisementRepository.java

```diff
@@ -43,6 +47,13 @@ Page<AdvertisementTitleProjection> findAllByIdNotAndSubcategoryId(@Param("id") U
     @Query("SELECT a FROM User u JOIN u.favoriteAdvertisements a WHERE u.username = :username")
     Page<AdvertisementTitleProjection> findFavoriteAdvertisementsByUsername(String username, Pageable pageable);
 
+    @Transactional
+    @Modifying
+    @Query(nativeQuery = true, value = "DELETE FROM favorite_advertisements a " +
+            "WHERE a.user_id = (select id from user where username = :username) " +
+            "and a.advertisement_id = :advertisementId")
+    int deleteFavoriteAdvertisementByAdvIdAndUsername(UUID advertisementId, String username);
```

Comment:
Did you try to construct the request using JPQL?
Kind of **removeAdvertisementByIdAndUserUsername**